src/libsyntax/parse/lexer/mod.rs

```diff
@@ -667,14 +666,9 @@ impl<'a> StringReader<'a> {
                     return None;
                 }
 
-                // I guess this is the only way to figure out if
-                // we're at the beginning of the file...
-                let smap = SourceMap::new(FilePathMapping::empty());
-                smap.files.borrow_mut().source_files.push(self.source_file.clone());
-                let loc = smap.lookup_char_pos_adj(self.pos);
-                debug!("Skipping a shebang");
-                if loc.line == 1 && loc.col == CharPos(0) {
-                    // FIXME: Add shebang "token", return it
+                let is_beginning_of_file = self.pos == self.source_file.start_pos;
```

Comment:
So, the new code accepts shebang at the start of anything that compiler treats as a source file - out-of-line `mod`, perhaps command line or macro?
And I'm not even sure what the old code does.
I rather wish this thing never existed in the language.

Reply:
My reading is that the old code did the same, but in a more elaborate way, by binary searching in a single-element slice. I’ve manually checked that shebangs are accepted  in the outofline modules today.

Reply:
For the referrnce, here’s the original commit that added the code: https://github.com/rust-lang/rust/pull/2430

Accidentally, it also added this _adj function, which I would like to get rid of next